Create release_management.md (#314)
DXHeroes/knowledge-base-content: practices/release_management.md

With correct release management, you can achieve: 

- Increases the success rate of releases while reducing releases with unexpected outcomes.
- Reduces unpredictable incidents caused by a release.
- Ability to manage all releases by creating a single documented process.
- Ability to record all releases by maintaining a single repository throughout the cycle.
- Compliance issues are escalated to management.
- Coordination between teams is improved to ensure services are delivered smoothly on schedule.
- Productivity is improved by establishing standard release tools and processes.
- Providing ample time for impact analysis by Change-Advisory Board (CAB).
- Documenting and establishing of auditable release controls is ensured.
- Communication of releases to the development team, client representatives, clients, and other organizations.
- Procedures are streamlined in order to achieve a balance between complexity and required controls of the release.
- Lessons learned from the release management process you can apply to other areas.

## Problems Release Management Solves

> - Unsuccessful product

Reviewer: ```suggestion
- [Unsuccessful product](/problems/unhappy-clients)
```

```

- Unhappy clients
- [Meaningless work](/problems/meaningless-work)
- [Poor code quality](/problems/poor-code-quality)
- [Increased cost](/problems/increased-cost)
- [Demotivated Team](/problems/demotivated-team)
- [Disconnect Between Business and IT](/problems/disconnect-between-business-and-it)
- [Bad product-market fit](/problems/bad-product-market-fit)

## How to Implement Release Management

1. **Plan the release**

>    Planning stage requires a lot of time for structuring the release from start to finish. A strong release plan will help teams stay on track and meet expected standards, aims and requirements.

Reviewer: ```suggestion
 The planning stage requires a lot of time for structuring the release from start to finish. A well-thought release plan will help teams stay on track and meet expected standards, aims, and requirements.
```

2. **Build the release**

>    This stage involves the actual development (designing and building) of the product according to requirements defined during the release plan. After dealing with any issues that surfaced, the build must be subjected to real-world scenario testing. As the development team builds the product, it is tested for user acceptance by using a testing environment. This helps the team find bugs or issues that can occur in the real world. If any issues are identified, the build is sent back for development. The work flow becomes cyclic as the build is sent from the 'development' stage to 'testing' stage and vice versa until the release is approved.

Reviewer: ```suggestion
This stage involves the actual development (designing and building) of the product according to requirements defined during the release plan. After dealing with any issues that surfaced, the build must be subjected to real-world scenario testing. As the development team builds the product, it is tested for user acceptance by using a testing environment. Testing helps the team find bugs or issues that can occur in the real world. As the testing team identifies errors they send the build back for development. The workflow becomes cyclic after the build moves the 'development' stage to the 'testing' stage and vice versa until the release is approved.
```

3. **User acceptance testing**

>    This involves the gathering of feedback from end users by allowing them to use an early version (free/beta/trial) of the product. This version is shared either online or with a large group of employees of a company. This is a vital step for the release management due to the large quantity of data gathered and fixes needed in order to prepare the product for its official launch.

Reviewer: ```suggestion
User Acceptance testing involves the gathering of feedback from end-users by allowing them to use an early version (alpha/beta/trial) of the product. Trials are shared either online or with a large group of employees of a company. Sharing is a vital step for the release management due to the large quantity of data gathered and fixes needed to prepare the product for its official launch.
```

4. **Prepare the release**

>    This stage involves the final changes done to the product after considering everything learned from User Acceptance Testing phase. Release preparation also includes a final quality report by the Quality Assurance (QA) team. During this time the QA team will conduct final checks to see if the build meets the minimum standards of acceptability and bussiness requirements which was defined in the release plan. Once this is completed, the functional team will validate findings and work on finalizing the release for deployment. The approval of the product manager is needed before deploying the build into a live environment.

Reviewer: ```suggestion
This stage involves the final changes done to the product after considering everything learned from the User Acceptance Testing phase. Release preparation also includes a final quality report by the Quality Assurance (QA) team. During this time, the QA team will conduct their checks to see if the build meets the minimum standards of acceptability and business requirements defined in the release plan. Afterward, the functional team will validate findings and work on finalizing the release for deployment. Before deploying the build into a live environment, the product manager should approve the changes.
```

5. **Deploy the release**

>    This is the final stage where the product is released to a live production environment. Apart from simply pushing out a build into production, this stage requires the need to educate the end users and your company. This can be achieved with the messaging, articles, documentation or demonstrations. This helps users to stay notified and updated on how to operate the new features of the release productively and efficiently. Finally, the development team should asess the performance of the release and discuss how to identify and document and issues to be addressed in the next iteration.

Reviewer: ```suggestion
Deployment is the final stage where the product release to a live production environment happens. Apart from pushing out a build into production, this stage requires the need to educate the end-users and your company. You can help it with the messaging, articles,  documentation, or demonstrations. Users should be notified and updated on how to operate the new features of the release productively and efficiently. Finally, the development team should assess the performance of the deployment and discuss how to identify and document and issues to be addressed in the next iteration.
```

## Common Pitfalls of Release Management

> 

Reviewer: We don't usually explain the pitfalls in our articles. Do you think you could rewrite them into single bullet point sentences?

- **Some strategies work better than others:**
  To have a better outcome, pitfalls of a product release must be examined and a more manageable release should be considered.

- **Coordinating the release:**

>   When designing the product, special thought must be given to its introduction, target user group, support offered and frequency of updates.

Reviewer: ```suggestion
 When designing the product, give a thought to its introduction, target user group, the support offered, and frequency of updates.
```

- **Conflict of user groups:**
  While one group prefers new features and updates the other may not. A strategy is needed to handle both groups and proceed with releases when needed.

- **Downtime for testing:**

>   When the release is tested either automatically, manually or both the product is not to be changed unless it is a bug fix.

Reviewer: ```suggestion
The product release cycle waits on bug fixes.
```

> - **Branching and Versioning:**

Reviewer: ```suggestion
- Lack of branch and version control.
```

---
slug: release-management
stages:
  - development
  - launch
short_description: Release management is the process of managing, planning, designing, scheduling, testing, controlling and deploying of a software build through different stages and environments; in preparation for software releases.
tags:
- release management
- workflow
- code coverage
- continuous integration
- continuous delivery
- feedback
- meaningless work
- technical writing
- software documentation
- introduction
- development
- code review
- mvp
- problems
- problem-solving
- assign
- leader
- responsibility
- versioning
keywords:
- release management
- workflow
- build
- deployment
- testing
- coordination
- branching
- versioning
---

# Release Management

**TL;DR**

> In the world of business and IT, software systems, resources and development processes  expand in distribution and complexity. How doas Release Management help this? Read more in this article.

Reviewer: ```suggestion
In the world of business and IT, software systems, resources, and development processes expand in distribution and complexity. How does Release Management help with controlling your deployments?

## Common Pitfalls of Release Management

- **Some strategies work better than others:**
  To have a better outcome, pitfalls of a product release must be examined and a more manageable release should be considered.

- **Coordinating the release:**
 When designing the product, give a thought to its introduction, target user group, the support offered, and frequency of updates.

- **Conflict of user groups:**
  While one group prefers new features and updates the other may not. A strategy is needed to handle both groups and proceed with releases when needed.

- **Downtime for testing:**
The product release cycle waits on bug fixes.

- Lack of branch and version control.
  A versioning system with multiple branching support helps managing features of the product. One branch maybe volatile due to new features being introduced while the other maybe stable.

## Resources for Release Management

> 

Reviewer: ```suggestion

- [Release Management Definition](https://en.wikipedia.org/wiki/Release_management)
```
